core/src/main/java/io/github/joblo2213/JMacros/core/config/ConfigManager.java

```diff
@@ -47,17 +47,23 @@
     private final List<ProfileData> profiles = new ArrayList<>();
     private Config config;
 
-    public ConfigManager(Path configDir) {
-        this.configDir = configDir;
-        this.configFile = configDir.resolve("config.json");
-        this.profilesDir = configDir.resolve("profiles");
-        this.iconsDir = configDir.resolve("icons");
-    }
-
     public ConfigManager(boolean isPortable) {
-        this(isPortable
-                ? Paths.get(System.getProperty("user.dir"))
-                : Paths.get(System.getProperty("user.home"), ".jmacros"));
+        if (isPortable) configDir = Paths.get(System.getProperty("user.dir"));
+        else {
+            Path configDir;
+            configDir = Paths.get(System.getProperty("user.home")).resolve(".jmacros");
+            if (System.getProperty("os.name").startsWith("mac"))
+                configDir = Paths.get(System.getProperty("user.home")).resolve("Library/Application Support/jmacros");
+
+            String xdgConfigHome = System.getenv().get("XDG_CONFIG_HOME");
+            if (!xdgConfigHome.isBlank()) configDir = Paths.get(xdgConfigHome).resolve("jmacros");
```

Comment:
If `XDG_CONFIG_HOME` environment variable is not defined this will throw a NullPointerException as `xdgConfigHome` will be null.

Also `XDG_CONFIG_HOME` may contain an invalid path name due to users messing with the env vars.  
We should catch those.

```suggestion
            String xdgConfigHome = System.getenv("XDG_CONFIG_HOME");
            try {
                if (xdgConfigHome != null && !xdgConfigHome.isBlank()) configDir = Paths.get(xdgConfigHome).resolve("jmacros");
            } catch (InvalidPathException e) {
                logger.error("XDG_CONFIG_HOME environment variable does not point to a valid path");
                logger.error(e.getMessage(), e);
                logger.info("Falling back to legacy dir");
            }
```